Add info on how to update a code snippet (#120)
Nexmo/station: lib/nexmo_developer/app/views/contribute/code-snippets/how-to-update-code-snippets.md

1. The actual code in the submodule repos (for example, https://github.com/Nexmo/nexmo-curl-code-snippets/blob/master/messages/messenger/send-text.sh)
2. The documentation pages that contain the tabbed code snippets in different programming languages (for example, https://developer.nexmo.com/messages/code-snippets/messenger/send-text)
3. The code snippet configuration (for example, `_examples/messages/messenger/send-text/curl.yml`)

Code snippets are described in more detail [here](/contribute/structure/guides/code-snippets).

## Updating code snippets

The tech writing team usually recommends coding and testing the code snippet code separately from NDP. This usually means you check out the relevant code snippet repo in a separate directory structure. This allows you to have local data such as private key, local config and so on for an app you can use for testing this repo. You can then run your test scripts locally to make sure everything works. See [Curl code snippet repo](https://github.com/Nexmo/nexmo-curl-code-snippets) for a good example of this set up.

1. Go to the code snippet repo directory and make your code changes in a branch. Test code changes locally.

> 2. If everything is OK, push up the branch to create a PR. Request a review. While waiting for a review you can work on the NDP side of things. Remember, as your changes at this point are on a PR you won’t affect NDP, so if, for example, you’ve deleted a code snippet, NDP won’t be affected - yet.

Reviewer: ```suggestion
2. If everything is OK, push up the branch to create a PR. Request a review. While waiting for a review you can work on the NDP side of things. Remember, as your changes have not been merged yet they will not be visible in  NDP, so if, for example, you’ve deleted a code snippet, NDP won’t be affected.
```

Here’s what some people do at this point - they merge their code snippet changes to master in the code snippet repo. If you deleted a code snippet or modified an existing one you just broke NDP - at least - those code snippet pages in NDP. The line number will be out, or worst case the code snippet has now gone completely. However, if you only added a code snippet you’re still good (although as yet the docs won't display it).

## Automatic NDP PR creation

> If you added a new code snippet and merged it into master of the code snippet repo, an NDP PR will **automatically** be created for you at this point. This is because at the least, NDP will need to be updated to point to the latest master of the code snippet repo, not the previous commit.

Reviewer: I think it would be helpful to have a note here to say that this is not always desirable. Certainly if you're only updating a single snippet or not affecting the display of snippets in NDP, it's a nice shortcut. Often however, you want to make sure that all the required changes are in place before creating a PR to point NDP at the latest version of the code snippet submodule repos. In that case, I usually delete the automatic update PRs with a note saying that I'll be pushing under a separate PR. (It doesn't always work though - for example, right now, someone merged the automatic PR for one of my snippet changes before I was done with the config)
